consensus/safety-rules/src/safety_rules.rs

```diff
@@ -109,6 +110,14 @@ impl SafetyRules {
 
         if self.decoupled_execution {
             Ok(vote_proposal.vote_data_ordering_only())
+            /*
+            // TODO: uncomment after we turn on extension check
+            let new_tree = vote_proposal
+                .accumulator_extension_proof()
+                .next_without_verify()
+                .map_err(|e| Error::InvalidAccumulatorExtension(e.to_string()))?;
+            Ok(vote_proposal.vote_data_with_extension_proof(&new_tree.0))
```

Comment:
even we do extension check it should not be here?

Reply:
we can put it here if we are doing a batch to batch extension check.